plugins/chat/assets/javascripts/discourse/components/chat/thread/header.gjs

```diff
@@ -24,14 +24,14 @@ export default class ChatThreadHeader extends Component {
       route = "chat.channel.threads";
       title = I18n.t("chat.return_to_threads_list");
       models = this.channel?.routeModels;
-    } else if (!this.currentUser.isInDoNotDisturb() && this.unreadCount > 0) {
```

Comment:
The order here has been changed to fix a navigation bug in the back button when we have thread notifications and we open a thread via My Threads (previously it would go to the channel which was weird).